src/Export/CacheHandler.php

```diff
@@ -0,0 +1,157 @@
+            if ($cacheItem) {
+                $cacheItem->set(true);
+                $cacheItem->expiresAfter(self::CACHE_LIFETIME_PRODUCT_GROUP);
+                $this->cache->save($cacheItem);
+            }
+        }
+    }
+
+    private function getDynamicProductGroupWarmedUpCacheItem(): CacheItemInterface
+    {
+        $id = sprintf('%s_%s_dynamic_product_warmup', self::CACHE_ID_PRODUCT_GROUP, $this->shopkey);
+
+        return $this->cache->getItem($id);
+    }
+
+    private function getDynamicProductGroupOffsetCacheItem(int $offset): CacheItemInterface
+    {
+        $id = sprintf('%s_%s_%s', self::CACHE_ID_PRODUCT_GROUP, $this->shopkey, $offset);
+
+        return $this->cache->getItem($id);
+    }
+
+    private function setTotalInCache(CacheItemInterface $cacheItem, $total): void
+    {
+        $cacheItem->set($total);
+        $cacheItem->expiresAfter(self::CACHE_LIFETIME_PRODUCT_GROUP);
+        $this->cache->save($cacheItem);
+    }
+
+    private function getDynamicGroupsTotalCacheItem(): CacheItemInterface
+    {
+        $id = sprintf('%s_%s_total', self::CACHE_ID_PRODUCT_GROUP, $this->shopkey);
+
+        return $this->cache->getItem($id);
+    }
+
+    /**
+     * Gets the total count of dynamic product groups from cache.
+     */
+    private function getDynamicProductGroupTotalFromCache(): int
+    {
+        $cacheItem = $this->getDynamicGroupsTotalCacheItem();
+        if ($cacheItem && $cacheItem->isHit()) {
+            $cacheItem->expiresAfter(self::CACHE_LIFETIME_PRODUCT_GROUP);
+            $this->cache->save($cacheItem);
+
+            return (int)$cacheItem->get();
+        }
+
+        return 0;
+    }
+
+    public function isDynamicProductGroupTotalCached(): bool
+    {
+        $totalCacheItem = $this->getDynamicGroupsTotalCacheItem();
+
+        return $totalCacheItem && $totalCacheItem->isHit();
+    }
+
+    public function setDynamicProductGroupTotal(int $total): void
+    {
+        $totalCacheItem = $this->getDynamicGroupsTotalCacheItem();
+        $this->setTotalInCache($totalCacheItem, $total);
+    }
+
+    public function setDynamicProductGroupOffset($products, int $offset): void
+    {
+        if ($offset > 0) {
+            $cacheItem = $this->getDynamicProductGroupOffsetCacheItem($offset);
+            $cacheItem->set($products);
+            $cacheItem->expiresAfter(self::CACHE_LIFETIME_PRODUCT_GROUP);
+            $this->cache->save($cacheItem);
+        }
+    }
+
+    public function isCacheWarmedUp(int $offset): bool
+    {
+        $cacheItem = $this->getDynamicProductGroupOffsetCacheItem($offset);
+        if ($cacheItem && $cacheItem->isHit()) {
+            $cacheItem->set(true);
+            $cacheItem->expiresAfter(self::CACHE_LIFETIME_PRODUCT_GROUP);
+            $this->cache->save($cacheItem);
+
+            return true;
+        }
+
+        return false;
+    }
+
+    public function isDynamicProductGroupsCached(): bool
```

Comment:
public and private methods are all floating around places. I usually prefer having public methods before private methods, in case the class has so many responsibilities as this class. This way the next developer don't has to scroll around and search the method he or she needs. Having such a distinction also helps getting an overall overview of what the class is responsible for.

Reply:
```suggestion
    public function areDynamicProductGroupsCached(): bool
```

Reply:
True but this was in relation to an already existing class. As this is a new class, please feel free to properly move the methods. I also don't expect many more review issues anyway at this point.